src/components/layout/Panel.tsx

```diff
@@ -93,7 +93,11 @@ export default function Panel({
       classes={heightConstraintClasses}
       elementRef={downcastRef(elementRef)}
     >
-      <CardHeader onClose={onClose} fullWidth={scrollable || fullWidthHeader}>
+      <CardHeader
+        onClose={onClose}
+        fullWidth={scrollable || fullWidthHeader}
+        size={paddingSize !== 'none' ? paddingSize : undefined}
```

Comment:
Inferring inner component props based on the main component props, which have an initial different purpose, could become confusing and produce unexpected results, as one could wonder why changing `paddingSize` affects other UI aspects.

If we want to be able to change the `Card`/`CardHeader` size when creating a `Panel`, perhaps it makes more sense to explicitly expose a `cardSize` prop.

Reply:
On top of that, if both components will need a matching size, perhaps it makes more sense to set it in the `Card` and make it expose it to children components via context.

Reply:
@acelaya I like your solution here of passing in a class when some property from the parent is `true`, implemented that here instead.